### DISCUSSION TITLE
How do I change the data location for MariaDB?

### DISCUSSION THREAD

**Discussion by julianrinaldi** [Homebrew/discussions#2222] [Everyday usage]
Everything I see online says to edit /Users/username/Library/LaunchAgents/homebrew.mxcl.mariadb.plist and change the datadir. However, whenever I start or restart MariaDB, it automatically reverts back to the previous information. Does anyone have any ideas? I'm trying to put the data on an external hard drive. The datadir location I'm using is /Volumes/Database/mysql. What am I doing wrong?

**Comment by SMillerDev** [comment]
Are you using `brew services`? Because that will regenerate the plist every time you start it. It would be a better choice to start mariadb manually and define the data directory that way.

**Reply by julianrinaldi** [reply]
Thanks. I actually just figured it out. The file in LaunchAgents is just an alias. So saving to it doesn't work. Modifying its source in /usr/local/Cellar/mariadb worked. Woops.

**Reply by SMillerDev** [reply]
That'll be overwritten with every update.
